### PR TITLE
Vet-clinic: Query and update database

### PR DESCRIPTION
**Pull request by Ceemos96**
### In this milestone, I : 
- Added a species column to the table
- Carried out multiple transactions and queries to the database.
- Add species column and set as "unspecified" Screenshot (58)
- Rollback changes on species column Screenshot (59)
- Set species for animals with names ending with 'mon' to be digimon Screenshot (60)
-  Set species for other animals with names not ending with 'mon' to be pokemon Screenshot (61)
- Delete animals table and rollback changes Screenshot (63)
- Carry out other transactions on the database
Screenshot (64)
Screenshot (65)
Screenshot (66)
- Query the database as requested in the project requirements Screenshot (67)

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ceemos, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO P1;
+UPDATE animals
+SET weight_kg = weight_kg * -1 WHERE SIGN(weight_kg) = -1;
+COMMIT;
+
+/* queries */
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
```
[the remark this answers is not in the document]
Comment (Ceemos96) [reply queries.sql:48]: The instructions state to query for:
-  The number of animals in the table.
- The number of animals that have never attempted to escape.

That is exactly what line 48 & 47 do.

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+UPDATE animals
+SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon' WHERE name NOT LIKE '%mon';
+COMMIT;
+
+/* transaction*/
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
[the remark this answers is not in the document]
Comment (Ceemos96) [reply queries.sql:33]: I confirmed the changes on the SQL command line. The instructions state that we add the changes to the queries.sql file that's why i added that.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+/* transaction*/
+BEGIN;
+UPDATE animals
+SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon' WHERE name NOT LIKE '%mon';
+COMMIT;
+
```
[the remark this answers is not in the document]
Comment (Ceemos96) [reply queries.sql:29]: I confirmed the changes on the SQL command line. The instructions state that we add the changes to the queries.sql file that's why i added that.

File: queries.sql [hunk 4]
```diff
@@ -8,3 +8,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+/* transaction*/
+ALTER TABLE animals
+ADD species char(45);
+
+BEGIN
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
```
[the remark this answers is not in the document]
Comment (Ceemos96) [reply queries.sql:21]: I confirmed the changes on the SQL command line. The instructions state that we add the changes to the queries.sql file that's why i added that.

File: schema.sql [hunk 5]
```diff
@@ -6,6 +6,7 @@ CREATE TABLE animals (
     date_of_birth date, 
     escape_attempts int, 
     neutered boolean, 
-    weight_kg decimal
-    
+    weight_kg decimal,
+    species char(45),
+    PRIMARY KEY (id)
     );
```
[the remark this answers is not in the document]
Comment (Ceemos96) [reply schema.sql:12]: I confirmed the changes on the SQL command line. The instructions state that we add the changes to the schema.sql file that's why i added that.

File: queries.sql [hunk 6]
```diff
@@ -8,3 +8,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+ROLLBACK TO P1;
+UPDATE animals
+SET weight_kg = weight_kg * -1 WHERE SIGN(weight_kg) = -1;
+COMMIT;
+
+/* queries */
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT name FROM animals WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```
[the remark this answers is not in the document]
Comment (Ceemos96) [reply queries.sql:50]: Hello. As seen in line 50, the database is queried for "who escaped the most". That is what the instruction states in the project requirement.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Ceemos96, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Ceemos96, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**